android/java/org/chromium/chrome/browser/crypto_wallet/fragments/BaseWalletNextPageFragment.java

```diff
@@ -129,4 +167,89 @@
             mAnimationDrawable.setExitFadeDuration(5000);
         }
     }
+
+    /** Shows biometric authentication button if supported and if it was previously set. */
+    protected void showBiometricAuthenticationButton(@NonNull final View view) {
+        if (Utils.isBiometricSupported(requireContext())
+                && KeystoreHelper.shouldUseBiometricToUnlock()) {
+            view.setVisibility(View.VISIBLE);
+        } else {
+            view.setVisibility(View.GONE);
+        }
+    }
+
+    @SuppressLint("MissingPermission")
+    @RequiresApi(api = Build.VERSION_CODES.P)
+    protected void showBiometricAuthenticationDialog(
+            @NonNull final View biometricUnlockButton,
+            @NonNull final BiometricAuthenticationCallback biometricAuthenticationCallback) {
+        final BiometricPrompt.AuthenticationCallback authenticationCallback =
+                new BiometricPrompt.AuthenticationCallback() {
+                    @Override
+                    public void onAuthenticationSucceeded(
```

Comment:
This code was moved to a base class to avoid duplication and left untouched, I thought it was already vetted by security team. 
I've filed a new GitHub issue to investigate if it's a false positive from the code scanning tool: https://github.com/brave/brave-browser/issues/40188

Reply:
Fingeprint authentication algorithm has a security review:
* https://github.com/brave/brave-core/pull/11298 
* https://github.com/brave/reviews/issues/677

Reply:
Is this a part of the onboarding flow? If yes than it's probably a false positive. At this stage we only set up the biometric authentication. Once it was set up, we will be using biometrics to access private key in the keychain, but we don't need to do it at the onboarding stage.

Reply:
oh nevermind, seems like we have an actual issue here. `authenticationResult` should be returning a secret that will decrypt wallet's password

Reply:
Created a PR to fix this: https://github.com/brave/brave-core/pull/25104